query/expression.go

```diff
@@ -46,34 +116,34 @@ type EvaluationContext struct {
 // Expressions correspond to the timerange in the current EvaluationContext.
 type Expression interface {
 	// Evaluate the given expression.
-	Evaluate(context EvaluationContext) (*api.SeriesList, error)
+	Evaluate(context EvaluationContext) (Value, error)
+}
+
+func EvaluateToSeriesList(e Expression, context EvaluationContext) (api.SeriesList, error) {
```

Comment:
Since it only uses public types and public methods/fields, even if it wasn't exported, an outside package could write it themselves...

Maybe some of the types should be private, like `Value`?